src/content/documentation/develop/firefox-builtin-data-consent.md

```diff
@@ -0,0 +1,248 @@
+
+### No data collection
+
+If an extension does not collect or transmit any data, developers should explicitly indicate that by specifying the `none` required permission in the manifest, as follows:
+
+```json
+{
+  "manifest_version": 2,
+  "name": "extension without data collection",
+  "version": "1.0.0",
+  "browser_specific_settings": {
+    "gecko": {
+      "id": "@extension-without-data-collection",
+      "data_collection_permissions": {
+        "required": ["none"]
+      }
+    }
+  },
+  "permissions": [
+    "bookmarks",
+    "<all_urls>"
+  ]
+}
+```
+
+When a user attempts to install this extension, Firefox will show the usual installation prompt with the description of the required (API) permissions as well as a new description to indicate that the extension does not collect any data (see Figure 3).
+
+![Figure 3: Installation prompt with no data transmission defined in the manifest](/assets/img/documentation/develop/data-collection-permissions-prompt-install-no-transmission.webp)
+*Figure 3: Installation prompt with no data transmission defined in the manifest*
+
+The "no data collected" type is also listed in the *Permissions and data* tab of the extension in `about:addons` as shown in Figure 4.
+
+![Figure 4: The "no data collected" permission is listed in about:addons](/assets/img/documentation/develop/data-collection-permissions-about-addons-no-transmission.webp)
+*Figure 4: The "no data collected" permission is listed in about:addons*
+
+Note: The `none` data type can only be required, and it cannot be used with other data types, including optional types. When that happens, Firefox will ignore the `none` type, and only consider the other data types (see next section for more information). In addition, Firefox will show a warning message intended to developers in `about:debugging` as shown in Figure 5.
+
+![Figure 5: A warning message is displayed when the none type is combined with other data collection permissions](/assets/img/documentation/develop/data-collection-none-type-warning.webp)
+*Figure 5: A warning message is displayed when the none type is combined with other data collection permissions*
+
+{% endcapture %}
+{% include modules/one-column.liquid,
+    id: "specifying-data-types"
+    content: content
+%}
+
+{% capture content %}
+
+## Accessing the data collection permissions programmatically
+
+Extension developers can use the [`browser.permissions` API](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/permissions) to interact with the optional data permissions. Specifically, the `getAll()` method returns the list of granted optional data permissions as follows:
+
+```js
+await browser.permissions.getAll()
+
+{
+  origins: ["<all_urls>"],
+  permissions: ["bookmarks"],
+  // In this case, the permission is granted.
+​  data_collection: ["technicalAndInteraction"]
+}
+```
+
```

Comment:
It could be useful to mention that the presence/absence of the `data_collection` in the `getAll()` response could be used to feature-detect the built-in data consent experience in Firefox at runtime.

```js
const perms = await browser.permissions.getAll();
if (!perms.data_collection) {
  // no built-in data consent in Firefox
}
```